sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -134,6 +131,32 @@ def fit(self, X, y=None):
 
         return self
 
+    def add_element(self, f, v, feature_names, vocab, fitting=True,
+                    transforming=False, indices=None, values=None):
+        if not isinstance(v, six.string_types) and isinstance(v, Iterable):
+            for vv in v:
```

Comment:
hmm.. it's not just a refactor but also address this  https://github.com/scikit-learn/scikit-learn/issues/6045

> I think it is convenient to accept list as a value of dict in DictVectorizer.fit()

and yeah. I'll add some benchmark data for this